src/services/videoconference/logic/server-helpers.js

```diff
@@ -0,0 +1,90 @@
+const { NotFound } = require('@feathersjs/errors');
+const { error } = require('../../../logger');
+const { ROLES } = require('./constants');
+const utils = require('./utils');
+
+const logErrorAndThrow = (message, response, ErrorClass = Error) => {
+	error(message, response);
+	throw new ErrorClass(message);
+};
+
+const createMeeting = async (server, id, name, params) => server.administration
+	.create(name, id, params)
+	.then((meeting = {}) => {
+		const { response } = meeting;
+
+		if (!utils.isValidFoundResponse(response)) {
+			logErrorAndThrow('meeting room creation failed', response);
+		}
+
+		return meeting;
+	});
+
+exports.ensureMeetingExists = async (server, id, name, params, create = false) => server.monitoring
+	.getMeetingInfo(id)
+	.then((meeting = {}) => {
+		const { response } = meeting;
+
+		if (utils.isValidNotFoundResponse(response)) {
+			if (create) {
+				return createMeeting(server, id, name, params);
+			}
+
+			return logErrorAndThrow('meeting room not found, create missing', response, NotFound);
+		}
+
+		return meeting;
+	});
```

Comment:
Here, I extracted the `createMeeting` to keep the function size sane. I also renamed the file from `logic/index.js` to `logic/server-helpers.js` as they're used in combination with the BBB wrapper in `logic/server.js`. Git marks all of this as new, but it's really just a small refactoring and file rename.